### PR TITLE
Improvements to Reporter labels

### LINKED ISSUE #1034: TableReporter does not write meaningful column labels
Currently, an output like this
```cpp
reporter->updInput("inputs").connect(myJoint->getCoordinateSet()[0].getOutput("value"));
```
produces a column labeled "value" in the resulting table. Better would be to use either the full path name or the annotation provided to `connect()` (if one is provided).

### PR DESCRIPTION
**Pull request by tkuchida**
ConsoleReporters now set labels using Input annotations, which can be specified by the user; TableReporters use full path names because, presumably, a user would want this detailed information in a file (and we are currently unable to determine the user's intent if they happen to specify an annotation that is the same as the default; added TODO in Reporter.h). Updates to hopper example and README.

Fixes #1034

### CODE REVIEW

**Comment by chrisdembia** [comment]
LGTM. +1 for remembering to update the README!

**Comment by chrisdembia** [comment]
@aseth1 might want to chime in, so I won't merge.

File: README.md [hunk 1]
```diff
@@ -104,7 +104,7 @@ This code produces the following animation:
 and prints the following information to the console:
 
         [reporter]
-                time|  fiber_force|       value|
+                time|  fiber_force| elbow_angle|
```
Comment (aseth1) [inline README.md]: Nice!

**Comment by aseth1** [comment]
LGTM

**Comment by aseth1** [comment]
Feel free to merge when tests pass.

**Comment by tkuchida** [comment]
Thanks for reviewing. All :white_check_mark:; merging.

**Pull request merged**